In [1]:
import pandas as pd


rA = pd.read_csv('realAll.csv')

In [2]:
rA = rA.rename(columns={'Unnamed: 0':'날짜'})
rA

날짜   월  사고수                   날씨
0     1   1    7                  NaN
1     2   1   16                  NaN
2     3   1   11               박무\n연무
3     4   1   15               박무\n연무
4     5   1    5               박무\n연무
..   ..  ..  ...                  ...
361  27  12    7                비\n박무
362  28  12    4                  달무리
363  29  12    9           비\n박무\n달무리
364  30  12   16  비\n눈\n진눈깨비\n소낙눈\n박무
365  31  12    7              소낙눈\n박무

[366 rows x 4 columns]

In [3]:
rA.to_csv('realAll.csv',index=False) #index = False를 해줘야 값만 보내짐.
rA

날짜   월  사고수                   날씨
0     1   1    7                  NaN
1     2   1   16                  NaN
2     3   1   11               박무\n연무
3     4   1   15               박무\n연무
4     5   1    5               박무\n연무
..   ..  ..  ...                  ...
361  27  12    7                비\n박무
362  28  12    4                  달무리
363  29  12    9           비\n박무\n달무리
364  30  12   16  비\n눈\n진눈깨비\n소낙눈\n박무
365  31  12    7              소낙눈\n박무

[366 rows x 4 columns]

In [4]:
rA.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 366 entries, 0 to 365
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   날짜      366 non-null    int64 
 1   월       366 non-null    int64 
 2   사고수     366 non-null    int64 
 3   날씨      231 non-null    object
dtypes: int64(3), object(1)
memory usage: 11.6+ KB


In [5]:
rA['사고수'].astype(int)
rA.sum()

C:\Users\BIT\AppData\Local\Temp/ipykernel_15748/3549867158.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  rA.sum()


날짜     5767
월      2384
사고수    4030
dtype: int64

In [6]:
rAw = rA.dropna()
rAw['사고수'].astype(int)
rAw.sum()

날짜                                                  3695
월                                                   1442
사고수                                                 2502
날씨     박무\n연무박무\n연무박무\n연무비비\n무지개비비비비\n박무비비\n연무연무비\n박무...
dtype: object

전체 = 4030

날씨 있는 날 사고수 = 2502
날씨 없는 날 사고수 = 1528

2020년은 366일. 
그중 날씨가 있는 날은 231일
없는 날은 135 일. 


### 오히려 

날씨가 있는 날은 10.83건으로 날씨가 없는 날보다 조금(0.48건 적다). (2502/231)

날씨가 없는 날 사고가 11.31일에 평균 109.25건  (1528/135)

제주도 역시 마찬가지로 날씨가 없는날 사고가 더 많이 난다.
그러나 정말 근소한 차이.

In [25]:
rA = rA.replace('(.*)\n(.*)', r'\1 \2', regex=True)
rA = rA.replace('(.*)눈(.*)', r'\1 1\2', regex=True)
rA = rA.replace('(.*)비(.*)', r'\1 2\2', regex=True)
rA = rA.replace('[^0-9]', '', regex=True)
rA = rA.replace('(.*)12(.*)', r'\1 3\2', regex=True)
rA = rA.replace('(.*)21(.*)', r'\1 3\2', regex=True) 

In [18]:
rA = rA.replace(' 3', '3', regex=True)

In [26]:
rA.head(10)

날짜  월  사고수   날씨
0   1  1    7  NaN
1   2  1   16  NaN
2   3  1   11     
3   4  1   15     
4   5  1    5     
5   6  1   18    2
6   7  1   10    2
7   8  1   13    2
8   9  1   14  NaN
9  10  1    9  NaN

In [10]:
rain = rA['날씨'] == '2'    # 비 만 온 날을 알수있다. 무언가 다를까 싶어서 해보았다.
b = rA[rain]
b

날짜   월  사고수 날씨
5     6   1   18  2
6     7   1   10  2
7     8   1   13  2
11   12   1    9  2
12   13   1   14  2
..   ..  ..  ... ..
355  21  12    5  2
357  23  12   12  2
358  24  12    6  2
361  27  12    7  2
363  29  12    9  2

[157 rows x 4 columns]

In [11]:
b.sum()

날짜                                                  2595
월                                                   1024
사고수                                                 1730
날씨     2222222222222222222222222222222222222222222222...
dtype: object

In [24]:
# 비가 확실히 많이 온다. 비온날 사고는 조금더 많은 편 
# 비오는날 157일 사고수 1730건 평균 11.01로 꽤 높은편 ㄷㄷ

In [13]:
snow = rA['날씨'] == '1'

a = rA[snow]


a

날짜   월  사고수 날씨
47   17   2   11  1
48   18   2    8  1
365  31  12    7  1

In [23]:
# 눈오는날 3일 사고수 26 건.

In [19]:
rsAll = rA['날씨'] == '3'

c = rA[rsAll]

c

날짜   월  사고수 날씨
35    5   2    5  3
46   16   2    9  3
348  14  12   10  3
349  15  12   12  3
350  16  12   13  3
364  30  12   16  3

In [21]:
c.sum()

날짜         96
월          52
사고수        65
날씨     333333
dtype: object

In [22]:
# 눈비 같이 오는날 6일 사고수 65건.